setup.test.js

```diff
@@ -83,13 +64,28 @@ describe("Imperial", () => {
             [Nation.AH, "d"],
             [Nation.IT, "e"],
             [Nation.GE, "f"],
-          ].map(([k, v]) => [
-            k,
-            { controller: v, treasury: 11, rondelPosition: null },
-          ])
+          ].map(([k, v]) => {
+            const nationBonds = [...AllBonds].filter(
+              (bond) =>
+                bond.nation === k && bond.number !== 4 && bond.number !== 1
+            );
```

Comment:
This code makes me think we should keep all available bonds in one place instead of distributing them among the nations. If we do that, we can instead write a test like this for each case:

```es6
describe('four players', () => { describe('available bonds', () => {
  expected = new Set([...AllBonds]) // clone AllBonds
  for (const n of Nation) {
    n.when({
      IT: () => {
        expected.delete(Bond(n, 1))
        expected.delete(Bond(n, 4))
      },
      FR: () => {
        // Nobody bought FR 1
        expected.delete(Bond(n, 4))
      },
      AH: () => {
        expected.delete(Bond(n, 1))
        expected.delete(Bond(n, 4))
      },
      GE: () => {
        expected.delete(Bond(n, 1))
        expected.delete(Bond(n, 4))
      },
      GB: () => {
        expected.delete(Bond(n, 1))
        // Nobody bought GB 4
      },
      RU: () => {
        // Nobody bought RU 1
        // Nobody bought RU 4
      },
    })
  }
  expect(actual.availableBonds).toEqual(expected)
})})
```

That is clearer in my opinion. We can also use this pool of bonds directly when we generate actions for a player based on bond cost:

```es6
[...game.availableBonds].filter(b => b.cost <= player.cash)
```

What do you think?